server/src/channels/commands/print_sysinfo.rs

```diff
@@ -0,0 +1,105 @@
+use crate::{
+    channels::server_command::ServerCommand, configs::server::ServerConfig,
+    streaming::systems::system::SharedSystem,
+};
+use async_trait::async_trait;
+use flume::{Receiver, Sender};
+use iggy::utils::duration::IggyDuration;
+use tokio::time::{self};
+use tracing::{error, info, warn};
+
+#[derive(Debug, Default, Clone)]
+pub struct SysInfoPrintCommand;
+
+pub struct SysInfoPrinter {
+    interval: IggyDuration,
+    sender: Sender<SysInfoPrintCommand>,
+}
+
+pub struct SysInfoPrintExecutor;
+
+impl SysInfoPrinter {
+    pub fn new(interval: IggyDuration, sender: Sender<SysInfoPrintCommand>) -> Self {
+        Self { interval, sender }
+    }
+
+    pub fn start(&self) {
+        let interval = self.interval;
+        let sender = self.sender.clone();
+        if interval.is_zero() {
+            info!("SysInfoPrinter is disabled.");
+            return;
+        }
+
+        info!(
+            "SysInfoPrinter is enabled, system information will be printed every {:?} seconds.",
+            interval.as_secs()
+        );
+
+        tokio::spawn(async move {
+            let mut interval_timer = time::interval(interval.get_duration());
+            loop {
+                interval_timer.tick().await;
+                let command = SysInfoPrintCommand {};
+                sender.send(command).unwrap_or_else(|error| {
+                    error!("Failed to send SysInfoPrintCommand. Error: {}", error);
+                });
+            }
+        });
+    }
+}
+
+#[async_trait]
+impl ServerCommand<SysInfoPrintCommand> for SysInfoPrintExecutor {
+    async fn execute(&mut self, system: &SharedSystem, _command: SysInfoPrintCommand) {
+        match system.read().get_sysinfo().await {
```

Comment:
We've discussed moving lock outside of the match block - is it ok to keep this read lock here?